### ISSUE TITLE
login does not handle responses other than 200 properly

### ISSUE THREAD

**Issue by Adyel** [wtekiela/opensub4j#10]
Hello,

I am trying this library for a class project. But I keep getting `java.lang.NullPointerException`. Don't know what's causing this. Can take a look?

`subtitles = osClient.searchSubtitles("eng", "Friends", "1", "1");` is causing the exception.

![image](https://user-images.githubusercontent.com/4653623/53289660-f8de1d80-37c2-11e9-986d-61e5643859e1.png)

Here is my code.

```java
        List<SubtitleInfo> subtitles;

        try {
            URL serverUrl = new URL("https", "api.opensubtitles.org", 443, "/xml-rpc");
            OpenSubtitlesClient osClient = new OpenSubtitlesClientImpl(serverUrl);
            osClient.login("myID", "andPassword", "en", "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0");
            System.out.println(osClient.isLoggedIn());
            subtitles = osClient.searchSubtitles("eng", "Friends", "1", "1");
            osClient.logout();
        } catch (MalformedURLException e) {
            e.printStackTrace();
        } catch (XmlRpcException e) {
            e.printStackTrace();
        }
```

Thanks a lot :smile:

**Comment by wtekiela** [comment]
Hi,
thanks for creating the issue. It seems that the library does not handle login exceptions very well, I'll work on that in the future (if I have time).

The problem in your example is that you are getting the following response from the server for the login method:
```
[DEBUG] RetriableXmlRpcClient - Calling method: LogIn, with params: [myID, andPassword, en, Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0]
[DEBUG] RetriableXmlRpcClient - Response: {seconds=0.014, token=042CtSkb7,zBOm-AXEa-w0hKJof, status=414 Unknown User Agent}
```
which essentially means that you are not logged in (_isLoggedIn_ should return false). You are using an improper user agent. Please read http://trac.opensubtitles.org/projects/opensubtitles/wiki/DevReadFirst for details. Excerpt from that page:

> ONLY for testing and developing purposes you can use special user agent "TemporaryUserAgent". Never use this UA as default/public/in production code, we periodically change this and your APP will stop work! Also don't abuse usage of this test user agent.

TL;DR: If you'll use TemporaryUserAgent in the login method, you'll be fine, e.g.
```
osClient.login( "en", "TemporaryUserAgent");
```

**Comment by wtekiela** [comment]
@Adyel this is fixed now and will be available when 0.2.2 is released.

Method `isLoggedIn()` will return `false` in your example.
Furthermore, you can investigate the response for login now in the following way:

```
Response response = osClient.login(...);
ResponseStatus = response.getStatus();
```

There is also an automated integration test for that: https://github.com/wtekiela/opensub4j/commit/8ee2a145b9db577011decc2530d1d5181353acbb

**Comment by wtekiela** [comment]
0.2.2 released https://repo1.maven.org/maven2/com/github/wtekiela/opensub4j/0.2.2/
